lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -1127,6 +1127,78 @@ LogicalResult InferenceMapping::mapOperation(Operation *op) {
         }
       })
 
+      // Handle memories.
+      .Case<MemOp>([&](auto op) {
+        // Create constraint variables for all ports.
+        for (auto result : op.results())
+          declareVars(result, op.getLoc());
+
+        // This creates independent variables for every data port. Yet, what we
+        // actually want is for all data ports to share the same variable. To do
+        // this, we find the first data port declared, and use that port's vars
+        // for all the other ports.
```

Comment:
The `MemOp` has a verifier to check that all data ports have the same type. This is a fine assumption.